### PR TITLE
264895 - Add register import status eventing

### PR DESCRIPTION
**Pull request by sfitz42**

### CODE REVIEW

File: app/config/storage/table.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const Joi = require('joi')
+
+const schema = Joi.object({
+  useConnectionString: Joi.bool().default(false),
+  connectionString: Joi.string().when('useConnectionString', { is: true, then: Joi.required(), otherwise: Joi.allow('').optional() }),
+  account: Joi.string().required(),
```
Comment (jbarnsley10) [inline app/config/storage/table.js:6]: A very picky comment but perhaps better to keep consistent naming with blob.js where 'account' is actually 'storageAccount'

File: app/storage/register-status-repository.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  })
+
+  return results
+}
+
+const setUploaded = async (filename, email) => await addToTable(filename, email, 'uploaded')
+
+const getLatestUpdate = async (filename) => {
+  const results = await queryImportsByPartition(filename, { top: 1 })
```
Comment (jbarnsley10) [inline app/storage/register-status-repository.js:49]: Will this not return the first row, not necessarily the latest update?

**Review by jbarnsley10 — DISMISSED** [review]
Great stuff. A couple of questions but more for me to understand than anything to change

File: app/storage/register-status-repository.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+const { odata } = require('@azure/data-tables')
+const { tableClient } = require('./get-table-client')
+
+const maxDate = new Date(8640000000000000)
+
+const invertTimestamp = (timestamp) => {
```
[the remark this answers is not in the document]
Comment (sfitz42) [reply app/storage/register-status-repository.js:6]: Table storage sorts all entities in a partition by the row key in ascending order. `8640000000000000` is the max date storable in JavaScript so by taking Date.now() from this value the latest record will always be returned first.

For the purposes of this feature, in pretty much all cases we only want the most recent import update.

See https://learn.microsoft.com/en-us/azure/storage/tables/table-storage-design-patterns#log-tail-pattern

File: app/storage/register-status-repository.js (hunk 2 above)
Comment (sfitz42) [reply app/storage/register-status-repository.js:49]: See above comment. Table storage doesn't provide a mechanism to orderBy so without the invertedTimestamp as row key we would have to get all entities in the partition and sort ourselves.

Using the log tail pattern, the `top: 1` will always return the smallest row key.

File: app/config/storage/table.js (hunk 1 above)
Comment (sfitz42) [reply app/config/storage/table.js]: Good point - just pushed an update.

**Review by jbarnsley10 — APPROVED** [review]

**Pull request merged**